### ISSUE TITLE
after updateing to 2.0.3 can't find backups anymore

### ISSUE THREAD

**Issue by yennor** [edvler/check_mk-urbackup-check#7]
I've updated the plugin to version 2.0.3. On both check mk server and clients.
I've got two urbackups server.
With one of them everything continued working. He continues finding the file and image backups.
On the other server he reported crashes in the plugin on the server side.
So i've decided to do a "remove all and find new" in the inventory, so he would discover all the services again.
He did that, but now he can't find any of file or image backups anymore.
While on the other working server, when I did a remove all and find new, everything worked as expected.

Here is the crash report I've got, before removing everything:

```
IndexError (list index out of range)

File "/omd/sites/rmm/lib/python3/cmk/base/agent_based/checking/_checking.py", line 413, in get_aggregated_result
    consume_check_results(
  File "/omd/sites/rmm/lib/python3/cmk/base/api/agent_based/checking_classes.py", line 494, in consume_check_results
    for subr in subresults:
  File "/omd/sites/rmm/lib/python3/cmk/base/api/agent_based/register/check_plugins.py", line 93, in filtered_generator
    for element in generator(*args, **kwargs):
  File "/omd/sites/rmm/local/lib/python3/cmk/base/plugins/agent_based/urbackup_check.py", line 145, in check_urbackup
    if check_name_urbackup(line[0]) == item:
  File "/omd/sites/rmm/local/lib/python3/cmk/base/plugins/agent_based/urbackup_check.py", line 121, in check_name_urbackup
    type_bkp = arr_backups[BACKUP_TYPE]



Local Variables | {'arr_backups': ['Vita'], 'check_line': 'Vita', 'machine_name': 'Vita'}
```

Agent output:

```
<<<urbackup_check>>>
Vita Server;;;;;filebackup;;;;;filebackup;;;;;2023-08-31_17:02+0200;;;;;True
Vita Server;;;;;imagebackup;;;;;No Imagebackup configured;;;;;0;;;;;False
hosting.example.com;;;;;filebackup;;;;;filebackup;;;;;2023-08-31_00:42+0200;;;;;True
hosting.example.com;;;;;imagebackup;;;;;No Imagebackup configured;;;;;0;;;;;False
intranet.example.com;;;;;filebackup;;;;;filebackup;;;;;2023-08-31_14:57+0200;;;;;True
intranet.example.com;;;;;imagebackup;;;;;No Imagebackup configured;;;;;0;;;;;False
work.example.com;;;;;filebackup;;;;;filebackup;;;;;2023-08-31_23:03+0200;;;;;True
work.example.com;;;;;imagebackup;;;;;No Imagebackup configured;;;;;0;;;;;False
<<<<Vita Server>>>>
<<<urbackup_check>>>
Vita Server;;;;;filebackup;;;;;filebackup;;;;;2023-08-31_17:02+0200;;;;;True
Vita Server;;;;;imagebackup;;;;;No Imagebackup configured;;;;;0;;;;;False
<<<<hosting.example.com>>>>
<<<urbackup_check>>>
hosting.example.com;;;;;filebackup;;;;;filebackup;;;;;2023-08-31_00:42+0200;;;;;True
hosting.example.com;;;;;imagebackup;;;;;No Imagebackup configured;;;;;0;;;;;False
<<<<intranet.example.com>>>>
<<<urbackup_check>>>
intranet.example.com;;;;;filebackup;;;;;filebackup;;;;;2023-08-31_14:57+0200;;;;;True
intranet.example.com;;;;;imagebackup;;;;;No Imagebackup configured;;;;;0;;;;;False
<<<<work.example.com>>>>
<<<urbackup_check>>>
work.example.com;;;;;filebackup;;;;;filebackup;;;;;2023-08-31_23:03+0200;;;;;True
work.example.com;;;;;imagebackup;;;;;No Imagebackup configured;;;;;0;;;;;False
```

**Comment by edvler** [comment]
Hello,
could you please test the version 2.0.4?

Greets
Matthias

**Comment by yennor** [comment]
unfortunately that doesn't work yet.
I'm getting still getting an error. It changed though.
But after looking at it i Guess the problem could be, that the Server name does have a space in it (Vita Server). Since in the local variables the machine name is called [vita]

```
IndexError (list index out of range)

  File "/omd/sites/rmm/lib/python3/cmk/base/modes/check_mk.py", line 1598, in mode_check_discovery
    check_result = discovery.execute_check_discovery(
  File "/omd/sites/rmm/lib/python3/cmk/base/agent_based/discovery/_discovery.py", line 119, in execute_check_discovery
    services = get_host_services(
  File "/omd/sites/rmm/lib/python3/cmk/base/agent_based/discovery/autodiscovery.py", line 593, in get_host_services
    **_get_node_services(
  File "/omd/sites/rmm/lib/python3/cmk/base/agent_based/discovery/autodiscovery.py", line 625, in _get_node_services
    service_result = analyse_discovered_services(
  File "/omd/sites/rmm/lib/python3/cmk/base/agent_based/discovery/_discovered_services.py", line 42, in analyse_discovered_services
    discovered_services=_discover_services(
  File "/omd/sites/rmm/lib/python3/cmk/base/agent_based/discovery/_discovered_services.py", line 150, in _discover_services
    {
  File "/omd/sites/rmm/lib/python3/cmk/base/agent_based/discovery/_discovered_services.py", line 150, in <dictcomp>
    {
  File "/omd/sites/rmm/lib/python3/cmk/base/agent_based/discovery/_discovered_services.py", line 303, in _discover_plugins_services
    yield from (
  File "/omd/sites/rmm/lib/python3/cmk/base/agent_based/discovery/_discovered_services.py", line 303, in <genexpr>
    yield from (
  File "/omd/sites/rmm/lib/python3/cmk/base/api/agent_based/register/check_plugins.py", line 73, in filtered_generator
    for element in generator(*args, **kwargs):
  File "/omd/sites/rmm/local/lib/python3/cmk/base/plugins/agent_based/urbackup_check.py", line 133, in discovery_urbackup_check
    yield Service(item=check_name_urbackup(line[0]), parameters=urbackup_check_default_levels)
  File "/omd/sites/rmm/local/lib/python3/cmk/base/plugins/agent_based/urbackup_check.py", line 121, in check_name_urbackup
    type_bkp = arr_backups[BACKUP_TYPE]


local variables:
{'arr_backups': ['Vita'], 'check_line': 'Vita', 'machine_name': 'Vita'}
```

**Comment by edvler** [comment]
Please check version 2.0.5. I think i should work now. It's indeed a problem with spaces in names, which I already fixed in version 2.0.4. But i missed one place in the code.

**Comment by yennor** [comment]
works like a charm. thanks a lot :-)
